lib/pf/UnifiedApi/Controller/Config/System.pm

```diff
@@ -0,0 +1,101 @@
+package pf::UnifiedApi::Controller::Config::System;
+
+=head1 NAME
+
+pf::UnifiedApi::Controller::Config::System -
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::UnifiedApi::Controller::Config::System
+
+=cut
+
+use strict;
+use warnings;
+use Mojo::Base 'pf::UnifiedApi::Controller::RestRoute';
+use pfappserver::Model::Enforcement;
+use pfappserver::Form::Interface::Create;
+use pf::UnifiedApi::Controller::Config;
+use pf::error qw(is_success);
+use Sys::Hostname;
+use pf::util;
+
+sub model {
+    require pfappserver::Model::Config::System;
+    return pfappserver::Model::Config::System->new();
+}
+
+sub get_gateway {
+    my ($self) = @_;
+    $self->render(json => {item => $self->model->getDefaultGateway()}, status => 200);
+}
+
+sub put_gateway {
+    my ($self) = @_;
+    require pfappserver::Model::Interface;
+    my $interfaces = pfappserver::Model::Interface->new->get('all');
+
+    my $gateway = $self->get_json ? $self->get_json->{gateway} : undef;
+    if($gateway) {
+        my ($status, $status_msg) = $self->model->write_network_persistent($interfaces, $gateway);
+        $self->render(json => {message => $status_msg}, status => $status);
+    }
+    else {
+        $self->render(json => {message => "Missing the gateway in the request payload"}, status => 422)
+    }
+}
+
+sub get_hostname {
+    my ($self) = @_;
+    $self->render(json => {item => hostname}, status => 200);
+}
+
+sub put_hostname {
+    my ($self) = @_;
+    my $hostname = $self->get_json ? $self->get_json->{hostname} : undef;
+    if($hostname) {
+        pf_run("hostnamectl set-hostname $hostname");
```

Comment:
Do we need hostnamectl in the sudoers file ?

Reply:
good catch, I was testing with `pfperl-api get` directly which is root in my case, I'll adjust the pf_run and add it to sudo